src/App.tsx

```diff
@@ -1,61 +1,166 @@
+import { useState } from 'react';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { ITodo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
 import './App.scss';
+import { IUser } from './types/User';
+
+const preperadTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: usersFromServer
+      .find(user => user.id === todo.userId) ?? null,
+  };
+});
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+interface ITodoSetting {
+  userText: string;
+  isUserTextEmpty: boolean;
+  selectedUser: IUser | null
+  isSubmit: boolean,
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<ITodo[]>(preperadTodos);
+
+  const initialTodoSetting = {
+    userText: '',
+    isUserTextEmpty: true,
+    selectedUser: null,
```

Comment:
i would change selected user to 0,  so in select you would have this 
`<option value="0" disabled>Choose a user</option>`
by default

so you wont need to check `value={selectedUser?.id ?? 0}` anything here and just have `value={selectedUser.id}` in your select tag

(but I'm not sure, maybe it would interfere your code logic somewhere else?)

Reply:
okay, i that selected user is an object, so with my approach you need to change everything, so don't mind this comment